lib/block-supports/background.php

```diff
@@ -53,10 +53,10 @@ function gutenberg_render_background_support( $block_content, $block ) {
 	}
 
 	$background_styles                    = array();
-	$background_styles['backgroundSize']  = isset( $block_attributes['style']['background']['backgroundSize'] ) ? $block_attributes['style']['background']['backgroundSize'] : 'cover';
 	$background_styles['backgroundImage'] = isset( $block_attributes['style']['background']['backgroundImage'] ) ? $block_attributes['style']['background']['backgroundImage'] : array();
 
-	if ( isset( $background_styles['backgroundImage']['source'] ) && 'file' === $background_styles['backgroundImage']['source'] && isset( $background_styles['backgroundImage']['url'] ) ) {
+	if ( ! empty( $background_styles['backgroundImage'] ) ) {
```

Comment:
Since we're not checking for the "source" prop anymore, and assuming (at least for now) that all values are paths to files, we can dispense with the extra checks.

Reply:
The "src" prop isn't required to render blocks.

For now the prop doesn't have any function in block attributes aside from being descriptive.

I _think_ that still okay - some of the [follow up tasks](https://github.com/WordPress/gutenberg/issues/54336) deal with other sources like featured images and, speculatively, hooking into openverse or allowing extenders to use filters to add sources. 

Featured images will be post 6.6 but it's slated and I reckon we'll need "src" as an identifier for that, so Gutenberg can fetch the post's featured image with it encounters `"src": "featured-image"` for example.

cc @andrewserong for brain check.

Reply:
My perspective on it is that I think it's fine to treat `"source": "file"` as the implicit default (the name `file` was borrowed from media links, from memory), but I still like the idea of it being stored by the block support in an explicit way, as it makes it clear how the `url` should be handled, and it's a natural place to flag featured images further down the track, as you mention.

As for this particular check, its original purpose was to ensure that other background rules (such as the fallback `cover` rule) are only output when an image is set. It looks like this check is now a bit more permissive, but is still in the same spirit of the check prior to this change, so sounds good to me!

Reply:
Yup, that makes sense!

Reply:
I'm removing the required file prop in a separate PR -  https://github.com/WordPress/gutenberg/pull/61387, similar to what's going on here but I thought it pretty useful.

It means that theme.json dev don't need to specific "source": "file" for image URLs, whereas at the moment they do.